### PR TITLE
WAS-373 Address validation

### PR DESCRIPTION
**Pull request by rickclare**

### CODE REVIEW

File: app/jobs/digital_services_core/uprn_address_populator_job.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        address_data = parser.address_data inbound
+
+        location.address.update!(address_data[0]) if address_data.try(:any?)
+      end
+    end
+
+    private
+
+    # this could probably go in an App specific initializer
+    # rubocop:disable Metrics/LineLength
```
Comment (Cruikshanks) [inline app/jobs/digital_services_core/uprn_address_populator_job.rb]: We've agreed 120 is our max for waste exemptions, and the same can apply to digital services core. Could you update .rubocop.yml to include that exception, then we wouldn't need this inline one here.

File: config/spring.rb [hunk 2]
```diff
@@ -0,0 +1,3 @@
+if defined?(Spring)
```
Comment (Cruikshanks) [inline config/spring.rb]: The commit history seems to show spring was added but then removed. Just checking that this is still valid.

File: spec/models/digital_services_core/user_spec.rb [hunk 3]
```diff
@@ -2,29 +2,20 @@
...
+        is_expected.to validate_presence_of :email
       end
 
-      it "has a password that does not meet our criteria", fail: true do
-        user.password = "123"
-        expect(user).not_to be_valid
-        expected = "is too short (minimum is 8 characters)"
-        expect(user.errors["password"]).to include(expected)
-      end
+      it { is_expected.to validate_length_of(:password).is_at_least(8) }
```
Comment (Cruikshanks) [inline spec/models/digital_services_core/user_spec.rb]: For this to be accurate, the > Password must be at least 8 characters and contain at least 1 uppercase letter, 1 lowercase letter and 1 number

Can we update this test to match that criteria.

**Comment by Cruikshanks** [comment]
Back to you @rickclare . Basically noddy comments to prove I did actually read the PR changes, but any issues with them shout.

File: config/spring.rb (hunk 2 above)
Comment (rickclare) [inline config/spring.rb]: Well spotted. I attempted to add _spring_, to speed up _rspec_ testing.  However, I couldn't get it working with Travis (Travis kept complaining, something to do with the "dummy" test engine, I think).  Therefore I removed it again.  I'll remove this _config/spring.rb_ file for completeness.

File: spec/models/digital_services_core/user_spec.rb (hunk 3 above)
Comment (rickclare) [inline spec/models/digital_services_core/user_spec.rb]: @Cruikshanks Pending test added as discussed: https://github.com/EnvironmentAgency/digital-services-core/commit/b9519c29fec8f50f2d3baf0c3e282b0f648709ae

**Comment by rickclare** [comment]
@Cruikshanks I've amended as per your comments, over to you for final review and merge.

**Pull request merged**